### PR TITLE
Add new rule: no-forbidden-elements

### LINKED ISSUE #747: add new rule: no-forbidden-elements
- disallow use of meta, style, html and script tags in template files. 
- rule should be configurable so teams can add their own disallowed elements

### PR DESCRIPTION
**Pull request by rajasegar**
- Disallow use of meta, style, html and script tags in template files.
- Rule should be configurable so teams can add their own disallowed elements

Fixes #747

### CODE REVIEW

File: lib/rules/no-forbidden-elements.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    );
+
+    throw new Error(errorMessage);
+  }
+  visitor() {
+    return {
+      ElementNode(node) {
+        let isForbiddenElement =
+          FORBIDDEN_ELEMENTS.includes(node.tag) || this.config.forbidden.includes(node.tag);
+        if (isForbiddenElement) {
```
Comment (rwjblue) [inline lib/rules/no-forbidden-elements.js:77]: We should not hard code `FORBIDDEN_ELEMENTS` like this, `this.config.forbidden` will include the `FORBIDDEN_ELEMENTS` by default (when using `true` for config value) but if configured the explicit list that is provided should be used. 

tldr;

```suggestion
        let isForbiddenElement = this.config.forbidden.includes(node.tag);
        if (isForbiddenElement) {
```

File: lib/rules/no-forbidden-elements.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+function ERROR_MESSAGE_FORBIDDEN_ELEMENTS(element) {
+  return `Use of <${element}> detected. Do not use forbidden elements.`;
+}
+
+const FORBIDDEN_ELEMENTS = ['meta', 'style', 'html', 'script'];
+
+const DEFAULT_CONFIG = {
+  forbidden: [],
+};
```
Comment (rwjblue) [inline lib/rules/no-forbidden-elements.js:14]: The `DEFAULT_CONFIG` should be where the list of default forbidden elements live:

```suggestion
const DEFAULT_CONFIG = {
  forbidden: ['meta', 'style', 'html', 'script'],
};
```

File: test/unit/rules/no-forbidden-elements-test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+const ERROR_MESSAGE_FORBIDDEN_ELEMENTS = require('../../../lib/rules/no-forbidden-elements')
+  .ERROR_MESSAGE_FORBIDDEN_ELEMENTS;
+
+generateRuleTests({
+  name: 'no-forbidden-elements',
+
+  config: true,
+
+  good: ['<header></header>', '<div></div>', '<footer></footer>', '<p></p>'],
```
Comment (rwjblue) [inline test/unit/rules/no-forbidden-elements-test.js:13]: Can you add some `good` examples that specify custom config to allow `<script>`?

Something like:

```js
{
  config: ['meta'],
  template: '<script></script>',
}
```

File: test/unit/rules/no-forbidden-elements-test.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  bad: [
```
Comment (rwjblue) [inline test/unit/rules/no-forbidden-elements-test.js:15]: Can you add `bad` examples where a user configures a custom element? Something like:

```
{
  config: ['Foo'],
  template: '<Foo />',
  result: {
    // error....
  }
}
```

File: docs/rule/no-forbidden-elements.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# no-forbidden-elements
+
+This rule disallows the use of `meta`, `style`, `html` and `script` tags in template files.
```
[the remark this answers is not in the document]
Comment (rajasegar) [reply docs/rule/no-forbidden-elements.md]: Done

File: lib/rules/no-forbidden-elements.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    } else if (!DEFAULT_CONFIG[key]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function sanitizeConfigArray(forbidden = []) {
+  return forbidden.filter((option) => option !== '').sort((a, b) => b.length - a.length);
```
[the remark this answers is not in the document]
Comment (rajasegar) [reply lib/rules/no-forbidden-elements.js]: Done

**Comment by rajasegar** [comment]
@rwjblue Addressed the review comments, please check and let me know.

File: test/unit/rules/no-forbidden-elements-test.js (hunk 4 above)
Comment (rajasegar) [reply test/unit/rules/no-forbidden-elements-test.js:23]: Added

File: test/unit/rules/no-forbidden-elements-test.js (hunk 3 above)
Comment (rajasegar) [reply test/unit/rules/no-forbidden-elements-test.js]: Done

File: lib/rules/no-forbidden-elements.js (hunk 2 above)
Comment (rajasegar) [reply lib/rules/no-forbidden-elements.js]: Done

File: lib/rules/no-forbidden-elements.js (hunk 1 above)
Comment (rajasegar) [reply lib/rules/no-forbidden-elements.js]: Done

File: lib/rules/no-forbidden-elements.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      config
+    );
+
+    throw new Error(errorMessage);
+  }
+  visitor() {
+    return {
+      ElementNode(node) {
+        let isForbiddenElement =
+          this.config.forbidden.includes(node.tag) || this.config.forbidden.includes(node.tag);
```
Comment (rwjblue) [inline lib/rules/no-forbidden-elements.js:76]: This condition seems odd to me, unless I'm mistaken this is checking the same thing twice?

Comment (rajasegar) [reply lib/rules/no-forbidden-elements.js]: Ah Yes, my bad, I applied your suggestion but failed to check the duplicates. Fixed it now in the latest commit
